File: schemas/booking-scheduling/drizzle/schedule_overrides.ts
// schedule_overrides: single-date exceptions to recurring schedule rules.
// See README.md for full design rationale.

import { pgTable, uuid, text, boolean, timestamp, index } from "drizzle-orm/pg-core";
import { schedules } from "./schedules";

export const scheduleOverrides = pgTable(
  "schedule_overrides",
  {
    id: uuid("id").primaryKey().defaultRandom(),
    scheduleId: uuid("schedule_id").notNull().references(() => schedules.id, { onDelete: "cascade" }),
    overrideDate: text("override_date").notNull(),
    startTime: text("start_time"),
    endTime: text("end_time"),
    isAvailable: boolean("is_available").notNull().default(true),
    reason: text("reason"),
    createdAt: timestamp("created_at", { withTimezone: true }).notNull().defaultNow(),
  },
  (table) => [
    index("idx_schedule_overrides_schedule_id_override_date").on(table.scheduleId, table.overrideDate),
  ]
);

File: schemas/booking-scheduling/spacetimedb/schedule_overrides.rs
// schedule_overrides: Date-specific availability overrides for a schedule.
// See README.md for full design rationale.

use spacetimedb::Timestamp;

#[spacetimedb::table(name = schedule_overrides, public)]
pub struct ScheduleOverride {
    #[primary_key]
    pub id: String, // UUID
    #[index(btree)]
    pub schedule_id: String, // UUID — FK → schedules.id (cascade delete)
    pub override_date: String,
    pub start_time: Option<String>,
    pub end_time: Option<String>,
    pub is_available: bool,
    pub reason: Option<String>,
    pub created_at: Timestamp,
    // Composite index: (schedule_id, override_date)
}
